plugins/catalog-backend/src/ingestion/processors/MicrosoftGraphOrgReaderProcessor.ts

```diff
@@ -58,6 +64,10 @@ export class MicrosoftGraphOrgReaderProcessor implements CatalogProcessor {
       return false;
     }
 
+    this.logger.warn(
+      'MicrosoftGraphOrgReaderProcessor from @backstage/plugin-catalog is deprecated and will be removed in the future. Please migrate to the new one from @backstage/plugin-catalog-backend-module-msgraph instead.',
```

Comment:
Was actually thinking that we remove this processor now and import the package instead. Either way we end up with the trouble of not having a clean way to replace already registered processors, making this pretty hard to do for a user.

Seems like we have a bit of an unfortunate dependency on us updating the catalog processor and builder API so that it's possible to replace processors, after which this is a reasonable ask, and at which point we can actually fully deprecate and remove the default processor.

Do you think it would be terrible to leave it in the slightly broken state of being really annoying to install the processor with a custom transform for now, knowing that that'll get cleaned up in the somewhat near future? What I'm proposing with that is that we simply remove the processor from the catalog backend and install the one from the module instead.

Reply:
So there will be a point in time where one has to migrate from automatic registration to manual registration for this processor. I don't think that we have to align both changes and we could actually remove it now, but with a good migration guide in the changeset.
If we just want to get rid of the duplicate code fast, that is the preferred way - I just remember the confusion around #6037 and #6045 😆 🤦 

What we can't do is a reexport, as this would result in a circular dependency of `@backstage/plugin-catalog-backend`.

I can offer to prepare a variant of this PR that removes it and adds a migration guide.

Reply:
Alright, breaking change + migration guide is probably best way to go for now, thanks 👍

Tbh the dependency on `@backstage/plugin-catalog-backend` from the processor is pretty unfortunate, we probably want to extract the processing dependencies into a separate package to avoid headaches in the future. Not as part of this PR ofc

Reply:
> Tbh the dependency on @backstage/plugin-catalog-backend from the processor is pretty unfortunate

Yep, I agree. Everything I need to write a custom processor (all the types and helpers) should be from something like `@backstage/plugin-catalog-node`.